### PR TITLE
Docs taxonomy support

### PR DESCRIPTION
**Pull request by jdonenine**
**What this PR does**:
* Adds the ability to provide `tags` and `versions` taxonomy to docs pages
* This was originally created as part of the prototyping of how to support k8ssandra-operator within the docs, and while we won't use this as our primary means, it's a good general feature addition that can be useful later
* Also updates our docs build scripts to support "staging" deployments (docs-staging.k8ssandra.io)
* Staging evolved from dev now that I have access to the cloudflare setup to manage the k8ssandra.io DNS -- it behaves in the same way that it will be updated on every push to the main branch.

This doesn't include any application of the taxonomy -- but here are some examples of what it will provide when used

On the sidebar you'll see a listing of all taxonomies:

Screen Shot 2022-01-04 at 4 17 04 PM

Screen Shot 2022-01-04 at 4 17 12 PM

It provides a navigational feature where you can click those and check out all of the pages that are associated with those taxonomies:

Screen Shot 2022-01-04 at 4 17 22 PM

Screen Shot 2022-01-04 at 4 17 40 PM

When you are on a doc page that includes these taxonomies are displayed as part of the page header:

Screen Shot 2022-01-04 at 4 17 53 PM

Screen Shot 2022-01-04 at 4 18 01 PM

This is all configured through the definition of each hugo page, something like this:

```
---
versions: ["K8ssandra v1", "K8ssandra v2"]
tags: ["Installation", "Get Started"]
title: "Install K8ssandra"
linkTitle: "Install"
weight: 3
description: >
  Topics for installing K8ssandra on local (dev) machines or production-level K8ssandra clusters on a variety of cloud platforms.
---
```

**Which issue(s) this PR fixes**:
Fixes N/A

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [x] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: docs/assets/scss/_taxonomy.scss [hunk 1]
```diff
@@ -0,0 +1,333 @@
+// Taxonomies - e.g. Tags, Categories, ...
```
Comment (jdonenine) [inline docs/assets/scss/_taxonomy.scss:1]: I did not create this file, it was brought over from Hugo's base set, just like the rest of the styling files that we have, it's just that it wasn't brought over when we initially built the site out.

File: docs/config/_default/config.toml [hunk 2]
```diff
@@ -16,7 +16,14 @@ defaultContentLanguageInSubdir = false
 # Useful when translating.
```
Comment (jdonenine) [inline docs/config/_default/config.toml:16]: Here with this file I've moved to a structure that allows hugo to override settings based on the "environment" selected when doing the building.

_default represents our "production" world -- the base set.

File: docs/config/dev/config.toml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+baseURL = "https://localhost:1313/"
```
Comment (jdonenine) [inline docs/config/dev/config.toml:1]: This (and the same in the staging file) allow for links to work properly across the app when not running the production site.

File: docs/layouts/_default/list.html [hunk 4]
```diff
@@ -2,9 +2,21 @@
 <div class="td-content">
 	<h1>{{ .Title }}</h1>
         {{ with .Params.description }}<div class="lead">{{ . | markdownify }}</div>{{ end }}
-	{{ if (and (not .Params.hide_readingtime) (.Site.Params.ui.readingtime.enable)) }}
-            {{ partial "reading-time.html" . }}
-        {{ end }}
+	<header class="article-meta">
```
Comment (jdonenine) [inline docs/layouts/_default/list.html:5]: The effort here (and for all of the `layouts` files here in this PR) was to migrate the content from the default hugo templates into ours -- when we created the site initially the templates were customized and the taxonomy pieces were taken out, this adds them back where necessary.  The end result is effectively the default hugo/docsy visualization of taxonomy.

**Review by johnsmartco — APPROVED** [review]
lgtm - thanks Jeff.

**Pull request merged**